src/query_plan/operation.rs

```diff
@@ -1269,6 +1380,40 @@ impl NormalizedFragmentSpreadSelection {
             unreachable!("We should always be able to either rebase the fragment spread OR throw an exception");
         }
     }
+
+    pub(crate) fn containment(
+        &self,
+        other: &NormalizedSelection,
+        options: ContainmentOptions,
+    ) -> Containment {
+        match other {
+            NormalizedSelection::Field(_) => Containment::NotContained,
+            NormalizedSelection::InlineFragment(other) => {
+                // Using keys here means that @defer fragments never compare equal.
+                // This is a bit odd but it is consistent: the selection set data structure would not
+                // even try to compare two @defer fragments, because their keys are different.
+                if self.spread.key() == other.inline_fragment.key() {
+                    self.selection_set
+                        .containment(&other.selection_set, options)
+                } else {
+                    Containment::NotContained
+                }
+            }
+            NormalizedSelection::FragmentSpread(other) => {
+                if self.spread.key() == other.spread.key() {
+                    self.selection_set
+                        .containment(&other.selection_set, options)
+                } else {
+                    Containment::NotContained
+                }
+            }
+        }
```

Comment:
There is a similar match statement around line 3620.

Reply:
Nice, you're right, I had this nesting initially because the condition was more complex and consisted of multiple statements, but now this can be merged.